samples/core/multi-device/main.c

```diff
@@ -0,0 +1,707 @@
+
+    // By default assume that there is only one device available.
+    // dev_opts->number is set to 1 so that when calling to cl_util_get_device
+    // for the second device there is no index out of range.
+    struct cl_sdk_options_SingleDevice dev_opts = {
+        .triplet = { 0, 0, CL_DEVICE_TYPE_ALL }
+    };
+    struct convolution_options convolution_opts = { .x_dim = 4096,
+                                                    .y_dim = 4096 };
+
+    OCLERROR_RET(
+        parse_options(argc, argv, &diag_opts, &dev_opts, &convolution_opts),
+        error, end);
+
+    // Create runtime objects based on user preference or default.
+    OCLERROR_PAR(dev = cl_util_get_device(dev_opts.triplet.plat_index,
+                                          dev_opts.triplet.dev_index,
+                                          dev_opts.triplet.dev_type, &error),
+                 error, end);
+
+    // Query OpenCL version supported by device.
+    char dev_version[64];
+    OCLERROR_RET(clGetDeviceInfo(dev, CL_DEVICE_VERSION, sizeof(dev_version),
+                                 &dev_version, NULL),
+                 error, end);
+
+    if (!diag_opts.quiet)
+    {
+        cl_util_print_device_info(dev);
+    }
+
+    if (diag_opts.verbose)
+    {
+        printf("Creating sub-devices...");
+        fflush(stdout);
+    }
+
+    if (opencl_version_contains(dev_version, "1.0")
+        || opencl_version_contains(dev_version, "1.1"))
+    {
+        fprintf(stdout,
+                "This sample requires device partitioning, which is an OpenCL "
+                "1.2 feature, but the device chosen only supports OpenCL %s. "
+                "Please try with a different OpenCL device instead.\n",
+                dev_version);
+        exit(EXIT_SUCCESS);
+    }
+
+    // Check if device supports fission.
+    cl_device_partition_property* dev_props = NULL;
+    size_t props_size = 0;
+    OCLERROR_RET(clGetDeviceInfo(dev, CL_DEVICE_PARTITION_PROPERTIES, 0, NULL,
+                                 &props_size),
+                 error, end);
+    if (props_size == 0)
+    {
+        fprintf(stdout,
+                "This sample requires device fission, which is a "
+                "feature available from OpenCL 1.2 on, but the "
+                "device chosen does not seem to support it. Please "
+                "try with a different OpenCL device instead.\n");
+        exit(EXIT_SUCCESS);
+    }
+
+    // Check if the "partition equally" type is supported.
+    MEM_CHECK(dev_props = (cl_device_partition_property*)malloc(sizeof(char)
+                                                                * props_size),
+              error, end);
+    OCLERROR_RET(clGetDeviceInfo(dev, CL_DEVICE_PARTITION_PROPERTIES,
+                                 sizeof(char) * props_size, dev_props, NULL),
+                 error, props);
+    size_t prop = 0;
+    for (; prop < props_size; ++prop)
```

Comment:
Shouldn't the limit be just: props_size / sizeof(cl_device_partition_property) ?

Reply:
Indeed. Fixed!